hack/update-bazel.sh

```diff
@@ -36,11 +36,16 @@ kube::util::go_install_from_commit \
 
 touch "${KUBE_ROOT}/vendor/BUILD"
 
-gazelle fix \
+(
+  # gazelle needs to run from the real ${KUBE_ROOT}, as otherwise its symlink
+  # following logic can get confused.
+  cd "${KUBE_ROOT}"
```

Comment:
Not sure I understand this - under what circumstances is it not?

Reply:
recent functionality in gazelle causes it to follow symlinks, but only if they're outside the repo root.

if you run `hack/run-in-gopath.sh hack/update-bazel.sh`, gazelle ends up running inside `_output/local/go/src/k8s.io/kubernetes`. gazelle doesn't currently resolve the repo root to an absolute path, so its [check to verify that the symlink is out-of-repo](https://github.com/bazelbuild/bazel-gazelle/blob/257c283f6dd0d177cad931f3e6897f07fed5b41e/internal/packages/walk.go#L452) fails (since it resolves symlinks its considering to their absolute path).

For example, consider `vendor/k8s.io/apimachinery`, which points to `../../staging/src/k8s.io/apimachinery`.

When run in the real `$KUBE_ROOT`, gazelle detects the repo root as `$HOME/go/src/k8s.io/kubernetes` and the endpoint of this symlink as `$HOME/go/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery`. Since that's within the repo, it doesn't follow the symlink.

When run through `hack/run-in-gopath.sh`, gazelle detects the repo root as `$HOME/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes`, but still resolves the endpoint of the vendor symlink as `$HOME/go/src/k8s.io/kubernetes/staging/src/k8s.io/apimachinery`. It considers this as out-of-repo and thus follows the symlink, erroneously.

I tried passing `-repo_root ${KUBE_ROOT}` instead of the chdir, but this was causing different problems in vendor/ that I haven't yet figured out.

I also have refrained from changing the cwd for `kazel`, since `kazel` will blow up if it's not run from a valid GOPATH.